storage/src/main/java/tech/pegasys/teku/storage/store/StoreVoteUpdater.java

```diff
@@ -92,10 +91,18 @@ public Bytes32 applyForkChoiceScoreChanges(
   public void commit() {
     // Votes are applied to the store immediately since the changes to the in-memory ProtoArray
     // can't be rolled back.
-    store.votes.putAll(votes);
-    store.highestVotedValidatorIndex =
-        store.highestVotedValidatorIndex.max(
-            votes.keySet().stream().max(Comparator.naturalOrder()).orElse(UInt64.ZERO));
+
+    store.highestVotedValidatorIndex = getHighestVotedValidatorIndex();
+
+    if (store.highestVotedValidatorIndex.intValue() >= store.votes.length) {
+      store.votes = Arrays.copyOf(store.votes, store.highestVotedValidatorIndex.intValue() + 1000);
```

Comment:
nit: I'd extract the `1000` here to a constant and then probably also ensure we leave that much empty space when first creating the `votes` array in the `Store` constructor as well.  Otherwise there's a good chance we'll have to resize the array immediately after startup when the first new validator activates.

Reply:
done